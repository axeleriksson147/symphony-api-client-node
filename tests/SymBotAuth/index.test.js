const SymBotAuth = require('../../lib/SymBotAuth');

jest.mock('fs');
jest.mock('https');
const mockFs = require('fs');
const mockHttps = require('https');

describe('SymBotAuth', () => {
  beforeEach(() => {
    mockFs.readFileSync = jest.fn((path) => "Contents of " + path);

    mockHttps.request = jest.fn(() => ({
      end: jest.fn(),
      on: jest.fn(),
      setTimeout: jest.fn()
    }));
  });

  afterEach(() => {
    jest.resetModules();
  });

  describe('sessionAuthenticate', () => {
    describe('when the auth type is not rsa', () => {
      test('POSTs to /sessionauth/v1/authenticate with the correct parameters', () => {
        SymBotAuth.sessionAuthenticate({
          botCertPath: "/",
          botCertName: "I am the cert",
          sessionAuthHost: "https://session.example.com",
          sessionAuthPort: 1234,
          botCertPassword: "Passphrase",
          podProxy: "agent"
        });

        expect(mockHttps.request.mock.calls).toMatchSnapshot();
      });
    });
  });

  describe('kmAuthenticate', () => {
    describe('when the auth type is not rsa', () => {
      test('POSTs to /keyauth/v1/authenticate with the correct parameters', () => {
        SymBotAuth.kmAuthenticate({
          botCertPath: "/",
          botCertName: "I am the cert",
          keyAuthHost: "https://key.example.com",
          keyAuthPort: 5678,
          botCertPassword: "Passphrase",
          keyManagerProxy: "agent"
        });

        expect(mockHttps.request.mock.calls).toMatchSnapshot();
      });
    });
  });
});
