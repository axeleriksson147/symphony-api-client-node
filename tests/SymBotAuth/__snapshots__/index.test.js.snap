// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SymBotAuth kmAuthenticate when the auth type is not rsa POSTs to /keyauth/v1/authenticate with the correct parameters 1`] = `
Array [
  Array [
    Object {
      "agent": "agent",
      "headers": Object {},
      "hostname": "https://key.example.com",
      "method": "POST",
      "passphrase": "Passphrase",
      "path": "/keyauth/v1/authenticate",
      "pfx": "Contents of /I am the cert",
      "port": 5678,
    },
    [Function],
  ],
]
`;

exports[`SymBotAuth sessionAuthenticate when the auth type is not rsa POSTs to /sessionauth/v1/authenticate with the correct parameters 1`] = `
Array [
  Array [
    Object {
      "agent": "agent",
      "headers": Object {},
      "hostname": "https://session.example.com",
      "method": "POST",
      "passphrase": "Passphrase",
      "path": "/sessionauth/v1/authenticate",
      "pfx": "Contents of /I am the cert",
      "port": 1234,
    },
    [Function],
  ],
]
`;
